torch/_inductor/autotune_process.py

```diff
@@ -201,14 +241,14 @@ def worker():
 
 
 def benchmark_in_sub_process(
-    choice: "ChoiceCaller",
+    choice: TritonTemplateCaller,
 ) -> float:
     """
     Do benchmarking in subprocess and return the perf number (latency).
     """
     assert choice.bmreq is not None
     tuning_process.initialize()
-    assert tuning_process.valid()
```

Comment:
This does not need to be transformed into a class method? All it needs is an isinstance assert so mypy can properly determine the type.

Reply:
Is this because attrs of class methods are propagated but normal methods are not? Regardless, the type hints on valid_process are wrong.

Reply:
(this part of the diff got reverted)